# Bonus: Temperature Analysis I

In [2]:
import pandas as pd
from datetime import datetime as dt
from datetime import date

In [3]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [4]:
# Convert the date column format from string to datetime
#convert the date string into datetime 
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [5]:
# Set the date column as the DataFrame index
df = df.set_index('date')
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

In [ ]:
# Drop the date column

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [7]:
# Add a column for the month of the year based on the index datetime for use later 
df['month'] = df.loc[:].index.month
df.head()

station  prcp  tobs  month
date                                      
2010-01-01  USC00519397  0.08    65      1
2010-01-02  USC00519397  0.00    63      1
2010-01-03  USC00519397  0.00    74      1
2010-01-04  USC00519397  0.00    76      1
2010-01-06  USC00519397   NaN    73      1

In [8]:
# Identify the average temperature for December and June 
ave_dec = df.loc[df['month']==12].mean()
ave_jun = df.loc[df['month']==6].mean() 
print(f"Mean June temp = {ave_jun['tobs']:.2f}F, Mean December temp = {ave_dec['tobs']:.2f}F")


Mean June temp = 74.94F, Mean December temp = 71.04F


In [15]:
# Create collections of the June and December observations 
dec_temp = df.loc[ df["month"] == 12,"tobs"]
jun_temp = df.loc[ df["month"] == 6,"tobs"]

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
Name: tobs, dtype: int64

In [11]:
# Run paired t-test
stats.ttest_ind(dec_temp,jun_temp)

Ttest_indResult(statistic=-31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis
We use the independent t-test since the numeber of observations is not necessarily equal for the two months. Test for the difference between the two means at the 0.05 level of significance.

H0 : The mean monthly temperature for June and December are equal  
H1 : The mean monthly temperature for June and December are not equal 

The pvalue = 3.9 10^-191 is (much) less than the significance level of 0.05 so we can reject the null hypothesis and conclude that there is a difference between the means.  

